### ISSUE TITLE
Should RCS1202 trigger on extension methods?

### ISSUE THREAD

**Issue by jnyrup** [JosefPihrt/Roslynator#597]
**Product and Version Used**: 
Visual Studio  16.2.4
Roslynator 2019 2.1.3

**Steps to Reproduce**:
In the following code RCS1202 is triggered on the invocation of `MyExtension`.
Is that a bug or intended behavior?

```c#
public interface IMyInterface
{
}

public class MyClass : IMyInterface
{
}

public static class MyClassExtensions
{
    public static int MyExtension(this MyClass _) => 42;
}

public class Class1
{
    public void Foo()
    {
        (GetInstanceOfInterface() as MyClass).MyExtension();
    }

    private static IMyInterface GetInstanceOfInterface() => null;
}
```

**Comment by clem0338** [comment]
Hi, looks like it does not work anymore:

`public static bool Let<T>(this T value, out T result) where T : class
{
    if(value != null) {
        result = value;
        return true;
    }

    result = default;
    return false;
}

if(order.Items.LastOrDefault().Let(out var lastItem)) {
// Some logic
}`

Regards

**Comment by JosefPihrt** [comment]
https://github.com/JosefPihrt/Roslynator/issues/790
